include/matx/transforms/svd/svd_cuda.h

```diff
@@ -755,72 +756,115 @@ using svd_cuda_cache_t = std::unordered_map<DnSVDCUDAParams_t, std::any, DnSVDCU
-  auto vt_col_maj = make_tensor<T1>(vt_new.Shape(), MATX_ASYNC_DEVICE_MEMORY, stream);
-
-  const char jobu_cusolver = detail::SVDJobToChar(jobu);
-  const char jobvt_cusolver = detail::SVDJobToChar(jobvt);
-
-  // Get parameters required by these tensors
-  auto params = detail::matxDnSVDCUDAPlan_t<decltype(u_new), decltype(s_new), decltype(vt_new), decltype(tvt)>::
-      GetSVDParams(u_col_maj, s_new, vt_col_maj, tvt, jobu_cusolver, jobvt_cusolver);
-
-  // Get cache or new QR plan if it doesn't exist
-  using cache_val_type = detail::matxDnSVDCUDAPlan_t<decltype(u_col_maj), decltype(s_new), decltype(vt_col_maj), decltype(tvt)>;
-  detail::GetCache().LookupAndExec<detail::svd_cuda_cache_t>(
-    detail::GetCacheIdFromType<detail::svd_cuda_cache_t>(),
-    params,
-    [&]() {
-      return std::make_shared<cache_val_type>(u_col_maj, s_new, vt_col_maj, tvt, jobu_cusolver, jobvt_cusolver);
-    },
-    [&](std::shared_ptr<cache_val_type> ctype) {
-      ctype->Exec(u_col_maj, s_new, vt_col_maj, tvt, exec, jobu_cusolver, jobvt_cusolver);
+  auto a_shape = a.Shape();
+  auto a_total_size = std::accumulate(a_shape.begin(), a_shape.begin() + ATensor::Rank(), 1, std::multiplies< typename ATensor::desc_type::shape_type>());
+  matxAlloc(reinterpret_cast<void **>(&tp), sizeof(T1) * a_total_size, MATX_ASYNC_DEVICE_MEMORY, stream);
+
+  const char job_cusolver = detail::SVDModeToChar(jobz);
+  const bool m_leq_n = a.Size(RANK-2) <= a.Size(RANK-1);
+  
+  if (m_leq_n) {
+    // get col-major AT
+    auto a_new = make_tensor(tp, a_shape);
+    (a_new = a).run(exec);
+    auto at_col_maj = transpose_matrix(a_new);
+
+    auto u_new = getSolverSupportedTensor(u, exec);
+    auto vt_new = getSolverSupportedTensor(vt, exec);
+
+    // swap U and VT
+    auto u_in = vt_new;
+    auto vt_in = u_new;
+
+    // Get parameters required by these tensors
+    auto params = detail::matxDnSVDCUDAPlan_t<decltype(u_in), decltype(s_new), decltype(vt_in), decltype(at_col_maj)>::
+      GetSVDParams(u_in, s_new, vt_in, at_col_maj, job_cusolver);
+
+    // Get cache or new SVD plan if it doesn't exist
+    using cache_val_type = detail::matxDnSVDCUDAPlan_t<decltype(u_in), decltype(s_new), decltype(vt_in), decltype(at_col_maj)>;
+    detail::GetCache().LookupAndExec<detail::svd_cuda_cache_t>(
+      detail::GetCacheIdFromType<detail::svd_cuda_cache_t>(),
+      params,
+      [&]() {
+        return std::make_shared<cache_val_type>(u_in, s_new, vt_in, at_col_maj, job_cusolver);
+      },
+      [&](std::shared_ptr<cache_val_type> ctype) {
+        ctype->Exec(u_in, s_new, vt_in, at_col_maj, exec, job_cusolver);
+      }
+    );
+
+    if(!u_new.isSameView(u)) {
+      (u = u_new).run(exec);
     }
-  );
+    if(!vt_new.isSameView(vt)) {
+      (vt = vt_new).run(exec);
+    }
+  } else {
+    // get col-major A
+    auto tv = TransposeCopy(tp, a, exec);
+    auto tvt = tv.PermuteMatrix();
+
+    auto u_col_maj = make_tensor<T1>(u.Shape(), MATX_ASYNC_DEVICE_MEMORY, stream);
```

Comment:
Above it uses `exec` to make a copy, but it seems like here it's assuming it's a stream. I think it's ok since we're in the CUDA file, but just making sure that's expected.

Reply:
Yea, I try to use `exec` when possible but `make_tensor` only supports a stream argument currently.